oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -82,21 +85,25 @@
      * value is {@code true} when at least an update is performed, otherwise {@code false}.
      */
     private final ConcurrentHashMap<Long, Boolean> updatesMap = new ConcurrentHashMap<>();
+    private final Set<String> failedDocSet = new HashSet<>();
     private final BulkProcessor bulkProcessor;
 
     ElasticIndexWriter(@NotNull ElasticConnection elasticConnection,
-                       @NotNull ElasticIndexDefinition indexDefinition) {
+                       @NotNull ElasticIndexDefinition indexDefinition, @NotNull NodeBuilder definitionBuilder) {
         this.elasticConnection = elasticConnection;
         this.indexDefinition = indexDefinition;
+        this.definitionBuilder = definitionBuilder;
         bulkProcessor = initBulkProcessor();
     }
 
     @TestOnly
     ElasticIndexWriter(@NotNull ElasticConnection elasticConnection,
                        @NotNull ElasticIndexDefinition indexDefinition,
-                       @NotNull BulkProcessor bulkProcessor) {
+                       @NotNull BulkProcessor bulkProcessor,
+                       @NotNull NodeBuilder definitionBuilder) {
         this.elasticConnection = elasticConnection;
         this.indexDefinition = indexDefinition;
+        this.definitionBuilder = definitionBuilder;
```

Comment:
NIT: `this.definitionBuilder = definitionBuilder;` should be after `this.bulkProcessor = bulkProcessor;`